### PR TITLE
[Android] Follow up fix for desktop mode option

### PR DESCRIPTION
**Pull request by samartnik**
Resolves https://github.com/brave/brave-browser/issues/20963

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Comment by samartnik** [comment]
Force pushed with lint fix
```
[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-android_desktop_mode_follow_up-noplatform/detail/pr-brave-browser-android_desktop_mode_follow_up-noplatform/1/pipeline#step-174-log-31)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-android_desktop_mode_follow_up-noplatform/detail/pr-brave-browser-android_desktop_mode_follow_up-noplatform/1/pipeline#step-174-log-32)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-android_desktop_mode_follow_up-noplatform/detail/pr-brave-browser-android_desktop_mode_follow_up-noplatform/1/pipeline#step-174-log-33)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-android_desktop_mode_follow_up-noplatform/detail/pr-brave-browser-android_desktop_mode_follow_up-noplatform/1/pipeline#step-174-log-34)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-android_desktop_mode_follow_up-noplatform/detail/pr-brave-browser-android_desktop_mode_follow_up-noplatform/1/pipeline#step-174-log-35)[](https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-android_desktop_mode_follow_up-noplatform/detail/pr-brave-browser-android_desktop_mode_follow_up-noplatform/1/pipeline#step-174-log-36)[2022-02-08T03:07:35.916Z] -private:

[2022-02-08T03:07:35.916Z] + private:

[2022-02-08T03:07:35.916Z]    bool need_override_ua_ = false;

[2022-02-08T03:07:35.916Z]    bool navigation_started_ = false;

[2022-02-08T03:07:35.916Z]  };

[2022-02-08T03:07:35.916Z] Format errors have been auto-fixed. Please review and commit these changes if lint was run locally. Otherwise run npm format to fix.

[2022-02-08T03:07:35.916Z] Format check failed.
```

File: browser/android/preferences/website/desktop_mode_tab_helper.cc [hunk 1]
```diff
@@ -46,4 +46,17 @@ void DesktopModeTabHelper::NavigationEntryCommitted(
...
+    content::NavigationHandle* navigation_handle) {
+  // We need to override UA on the first navigation only.
+  if (!navigation_started_) {
+    navigation_started_ = true;
+    need_override_ua_ = true;
+    return;
+  }
+
+  // We should not override UA on all the next navigations.
+  need_override_ua_ = false;
```
Comment (simonhong) [inline browser/android/preferences/website/desktop_mode_tab_helper.cc:59]: Hmm, I think this PR's changes seems same with initial version as first navigation doesn't have last committed entry?
```
  content::NavigationEntry* entry =
      web_contents()->GetController().GetLastCommittedEntry();
  need_override_ua_ = entry == nullptr;
```

Comment (samartnik) [reply browser/android/preferences/website/desktop_mode_tab_helper.cc:59]: Somehow it does, with `need_override_ua_ = entry == nullptr` `need_override_ua_` is always `false` in `cr98`

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Pixel 6` running `Android 12` using 

**`Desktop Mode Disabled`**

* ensured that tapping on `Desktop site` when `Desktop Mode` is disabled correctly loaded the desktop version of the website
  * ensured when enabled, the checkmark is correctly visible indicating that `Desktop site` is being used
  * ensured that tapping it again removes the checkmark under `Desktop site` and loads the mobile version of the website
* ensured `Desktop site` is reset back to disabled if it was enabled and the tab was closed as `Desktop Mode` is disabled.

**`Desktop Mode Enabled via Site Settings`**

* ensured that websites are being loaded in desktop view by default and `Desktop site` has a checkmark indicating it's enabled 
* ensured that tapping on `Desktop site` when `Desktop Mode` is enabled correctly loaded the mobile version of the website
  * ensure that the checkmark is removed from `Desktop site` to indicate that it's disabled for that particular website
* ensured that closing/re-opening the website that was being viewed via `mobile` view is restored into desktop view as `Desktop Mode` is enabled

**Pull request merged**